doc/sphinx/proofs/writing-proofs/rewriting.rst

```diff
@@ -384,297 +381,248 @@ Rewriting with definitional equality
- on transparency and opacity).
+ The δ rule only applies to transparent constants
+ (see :ref:`controlling-the-reduction-strategies` on transparency and opacity).
+
+.. tacn:: simpl {? @delta_reductions } {? {| @reference_occs | @pattern_occs } } {? @occurrences }
 
-.. tacn:: cbn
-          simpl
-   :name: cbn; simpl
+   .. insertprodn reference_occs pattern_occs
 
-   These tactics apply to any goal. They try to reduce a term to
-   something still readable instead of fully normalizing it. They perform
+   .. prodn::
+      reference_occs ::= @reference {? at @occs_nums }
+      pattern_occs ::= @one_term {? at @occs_nums }
+
+   Reduces a term to
+   something still readable instead of fully normalizing it. It performs
    a sort of strong normalization with two key differences:
 
-   + They unfold a constant if and only if it leads to a :math:`\iota`-reduction,
+   + It unfolds constants only if they lead to an ι-reduction,
      i.e. reducing a match or unfolding a fixpoint.
-   + While reducing a constant unfolding to (co)fixpoints, the tactics
-     use the name of the constant the (co)fixpoint comes from instead of
+   + When reducing a constant unfolding to (co)fixpoints, the tactic
+     uses the name of the constant the (co)fixpoint comes from instead of
      the (co)fixpoint definition in recursive calls.
 
-   The :tacn:`cbn` tactic was intended to be a more principled, faster and more
-   predictable replacement for :tacn:`simpl`.
-
-   The :tacn:`cbn` tactic accepts the same flags as :tacn:`cbv` and
-   :tacn:`lazy`. The behavior of both :tacn:`simpl` and :tacn:`cbn`
-   can be tuned using the :cmd:`Arguments` command.
-
-   .. todo add "See <subsection about controlling the behavior of reduction strategies>"
-      to TBA section
+   :n:`@occurrences`
+     Permits selecting whether to reduce the conclusion and/or one or more
+     hypotheses.  The `at` option of :n:`@occurrences` is not allowed here,
+     but :n:`reference_occs` and :n:`@pattern_occs` have a somewhat less
+     flexible `at` option for selecting specific occurrences.
 
-   Notice that only transparent constants whose name can be reused in the
-   recursive calls are possibly unfolded by :tacn:`simpl`. For instance a
-   constant defined by :g:`plus' := plus` is possibly unfolded and reused in
-   the recursive calls, but a constant such as :g:`succ := plus (S O)` is
-   never unfolded. This is the main difference between :tacn:`simpl` and :tacn:`cbn`.
-   The tactic :tacn:`cbn` reduces whenever it will be able to reuse it or not:
-   :g:`succ t` is reduced to :g:`S t`.
+   :tacn:`simpl` can unfold transparent constants whose name can be reused in
+   recursive calls as well as those designated by :cmd:`Arguments` :n:`@reference … /`
+   commands. For instance, a constant :g:`plus' := plus` may be unfolded and
+   reused in recursive calls, but a constant such as :g:`succ := plus (S O)` is
+   never unfolded.
```

Comment:
```suggestion
   not unfolded unless it was specifically designated in a call to :cmd:`Arguments`.
```

Example:

```coq
Definition succ := plus (S O).

Eval simpl in succ. (*
     = succ
     : nat -> nat
*)

Arguments succ /.

Eval simpl in succ. (*
     = fun m : nat => S m
     : nat -> nat
*)
```

Reply:
I expanded on your wording a little.  Didn't add the example.

Reply:
Yep, the example was not meant to be included.